src/octoclass.cpp

```diff
@@ -1112,6 +1153,34 @@ bool OctoClass::NearestOccNodeWithinRadius(const geometry_msgs::Point &center_pt
     return there_are_nodes;
 }
 
+bool OctoClass::NearestOccNodeWithinBox(const octomap::point3d &center_pt,
+                                        const double &box_half_width,
+                                        double *distance) {
+    const Eigen::Vector3d center(center_pt.x(), center_pt.y(), center_pt.z());
+    const Eigen::Vector3d box_min = center - Eigen::Vector3d(box_half_width, box_half_width, box_half_width);
+    const Eigen::Vector3d box_max = center + Eigen::Vector3d(box_half_width, box_half_width, box_half_width);
+    std::vector<Eigen::Vector3d> candidates;
+    std::vector<double> node_sizes;
+
+    // Initialize output
+    *distance = std::numeric_limits<double>::infinity();
+
+    // get all occupied nodes within a box
+    this->OccNodesWithinBox(box_min, box_max, &candidates, &node_sizes);
+
+    // Check if any of the candidates are within radius
+    bool there_are_nodes = false;
+    for (uint i = 0; i < candidates.size(); i++) {
+        const Eigen::Vector3d dist_vec = candidates[i] - center;
+        if (dist_vec.norm() < *distance) {
+            *distance = dist_vec.norm();
```

Comment:
norm() is calculated twice on each loop here - could be made faster easily

Reply:
Good catch! I did some extra improvements by only computing the square root if the evaluated not is indeed closer